webapp/src/components/cardDialog.tsx

```diff
@@ -124,14 +151,17 @@ const CardDialog = (props: Props): JSX.Element => {
                     readonly={props.readonly}
                 />}
 
-            {!card &&
+                {!card &&
                 <div className='banner error'>
                     <FormattedMessage
                         id='CardDialog.nocard'
                         defaultMessage="This card doesn't exist or is inaccessible."
                     />
                 </div>}
-        </Dialog>
+            </Dialog>
+
+            {showConfirmationDialogBox && <ConfirmationDialogBox dialogBox={confirmDialogProps}/>}
```

Comment:
State is needed. I set confirmDialogProps when showing Dialog.

Reply:
In both `cardDialog.tsx` and `kanbanCard.tsx`, the state for confirmDialogProps never changes it is always the same. It could actually be set to this value by default. Therefore, storing this in state is unnecessary. Simply create the properties struct when you call `<ConfirmationDialogBox>` or create a variable to hold those properties and use it here. 

```
{showConfirmationDialogBox && 
    <ConfirmationDialogBox dialogBox={
        heading: intl.formatMessage({id: 'CardDialog.delete-confirmation-dialog-heading', defaultMessage: 'Confirm card delete!'}),
        confirmButtonText: intl.formatMessage({id: 'CardDialog.delete-confirmation-dialog-button-text', defaultMessage: 'Delete'}),
        onConfirm: handleDeleteCard,
        onClose: () => {
            setShowConfirmationDialogBox(false)
        },
    }/>
}
```

Let me know if I'm not understanding something. Thanks